parser_library/src/processing/preprocessors/cics_preprocessor.cpp

```diff
@@ -1110,7 +1110,8 @@ class cics_preprocessor final : public preprocessor
         const preprocessor::line_iterator& end,
         const std::optional<size_t>& potential_lineno)
     {
-        static const std::regex exec_cics("^([^ ]*)[ ]+([eE][xX][eE][cC][ ]+[cC][iI][cC][sS](?:[ ]+(\\S+))?)(?= |$)");
+        static const std::regex exec_cics(
+            "^([^ ]*)[ ]+(?:([eE][xX][eE][cC][ ]+[cC][iI][cC][sS])(?:[ ]+(\\S+))?)(?= |$)");
```

Comment:
is the first non-capture group actually needed?

Reply:
The following test doesn't work without the non-capturing group:
https://github.com/BroadcomMFD-DevOps/che-che4z-lsp-for-hlasm/blob/0ccd11d675fd3f4dc4c2b76a7391ffbc90cf1353/parser_library/test/processing/cics_preprocessor_test.cpp#L483

Reply:
Actually, seems like I removed wrong parenthesis and the test does indeed pass with this regex: 
`^([^ ]*)[ ]+([eE][xX][eE][cC][ ]+[cC][iI][cC][sS])(?:[ ]+(\\S+))?(?= |$)`

Reply:
This part `(?:([eE][xX]`, not `(?:[ ]+(\\S+))?`

Reply:
Yup, I meant that I removed wrong _**end**_ parenthesis and didn't notice at first.